src/components/map/CountriesTileLayer.js

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState } from "react";
+import { useMap } from "react-leaflet";
+import L from 'leaflet';
+import { layerStyle, highlightStyles } from './MapStyle';
+import _ from "lodash";
+import { createAltPopup } from "../../utils/mapUtils"
+import ReactDOMServer from 'react-dom/server';
+// eslint-disable-next-line no-unused-vars
+import VectorGrid from "leaflet.vectorgrid";
+
+const toggleCountrySelection = (e, layer, highlightStyle) => {
+  // only change selection style on countries that have seroprev data
+  const layerStyle = layer._overriddenStyles[e.layer.properties.CODE]; //TODO: refactor this kinda ugly
+
+  if (layerStyle) 
+    layer.setFeatureStyle(e.layer.properties.CODE, highlightStyle);
+}
+
+export default function CountriesTileLayer(props) {
+
+  const { url, records, zIndex } = props;
+  const map = useMap();
+  const [layer, setLayer] = useState(undefined);
+  
+  useEffect(() => {
+        
+    var layer = L.vectorGrid.protobuf(`${url}/tile/{z}/{y}/{x}.pbf`, {
```

Comment:
should make this a const?

Reply:
Also maybe call this something other than layer so that we don't confuse this variable with the "layer" variable defined by useState

Reply:
we should make this a const

Reply:
fixed & renamed